packages/block-library/src/comments/index.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Server-side rendering fallback of the `core/comments-query-loop` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-query-loop` block fallback on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Returns the filtered post comments for the current post wrapped inside "p" tags.
+ */
+function render_block_core_comments_query_loop( $attributes, $content, $block ) {
+	global $post;
+
+	$is_legacy = 'core/post-comments' === $block->name || isset( $attributes['legacy'] );
+	if ( ! $is_legacy ) {
+		$inner_blocks_html = '';
+		foreach ( $block->inner_blocks as $inner_block ) {
+			$inner_blocks_html .= $inner_block->render();
+		}
+		return $inner_blocks_html;
+	}
+
+	$post_id = $block->context['postId'];
+	if ( ! isset( $post_id ) ) {
+		return '';
+	}
+
+	$comment_args = array(
+		'post_id' => $post_id,
+		'count'   => true,
+	);
+	// Return early if there are no comments and comments are closed.
+	if ( ! comments_open( $post_id ) && get_comments( $comment_args ) === 0 ) {
+		return '';
+	}
+
+	$post_before = $post;
+	$post        = get_post( $post_id );
+	setup_postdata( $post );
+
+	ob_start();
+	// There's a deprecation warning generated by WP Core.
+	// Ideally this deprecation is removed from Core.
+	// In the meantime, this removes it from the output.
+	add_filter( 'deprecated_file_trigger_error', '__return_false' );
+	comments_template();
+	remove_filter( 'deprecated_file_trigger_error', '__return_false' );
+	$post = $post_before;
+
+	$classnames = array();
+	// Adds the old class name for styles' backwards compatibility.
+	if ( isset( $attributes['legacy'] ) ) {
+		$classnames[] = 'wp-block-post-comments';
+	}
+	if ( isset( $attributes['textAlign'] ) ) {
+		$classnames[] = 'has-text-align-' . $attributes['textAlign'];
+	}
+
+	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => implode(' ', $classnames ) ) );
+	$output             = ob_get_clean();
+
+	wp_enqueue_script( 'comment-reply' );
+	enqueue_legacy_post_comments_block_styles( $block->name );
+
+	return sprintf( '<div %1$s>%2$s</div>', $wrapper_attributes, $output );
+}
+
+/**
+ * Registers the `core/comments-query-loop` block on the server.
```

Comment:
```suggestion
 * Registers the `core/comments` block on the server.
```

...and a few more below 😅

Reply:
And fixed 😊